tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```diff
@@ -0,0 +1,740 @@
+        count.getAndIncrement();
+    }
+
+    /**
+     * Copies a Descriptor into a temporary one for it can be cloned into the list of descriptors
+     *
+     * @param original The original descriptor that is continually modified
+     */
+    private void copyAddResetDescriptor(Descriptor original) {
+        Descriptor temp = new Descriptor(original);
+        descriptors.add(temp);
+        original.reset();
+    }
+
+    /**
+     * Determines if the current descriptor block is missing any required values
+     * and then if it is, prints out a warning message and sets that required value
+     * and the optional values to null
+     *
+     * @param count The index of the current descriptor block
+     */
+    private void findMissingRequiredValuesAndSetValues(int count, Descriptor descriptor) {
+        boolean missingRequiredValue = false;
+        if (descriptor.getInputRealmLocation() == null) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlock(count, "missing input realm location"));
+            missingRequiredValue = true;
+        }
+        if (descriptor.getOutputRealmLocation() == null) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlock(count, "missing output realm location"));
+            missingRequiredValue = true;
+        }
+        if (descriptor.getFileSystemRealmName() == null) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlock(count, "missing new filesystem realm name"));
+            missingRequiredValue = true;
+        }
+        if(descriptor.getHashCharset() == null) {
+            descriptor.setHashCharset(StandardCharsets.UTF_8);
+        }
+        if(descriptor.getHashEncoding() == null) {
+            descriptor.setHashEncoding(Encoding.BASE64);
+        }
+        if(descriptor.getEncoded() == null) {
+            descriptor.setEncoded(true);
+        }
+        if(descriptor.getLevels() == 0) {
+            descriptor.setLevels(2);
+        }
+        if (missingRequiredValue) {
+            descriptor.reset();
+        }
+    }
+
+    /**
+     * Handles creating the Elytron filesystem-realm from the descriptor array
+     *
+     * @throws Exception Exception to be handled by Elytron Tool
+     */
+    private void createFileSystemRealm() throws Exception {
+        Security.addProvider(new WildFlyElytronProvider());
+        for (Descriptor descriptor : descriptors) {
+            System.out.println("Creating encrypted realm for: " + descriptor.getInputRealmLocation());
+            if (descriptor.getFileSystemRealmName() == null || descriptor.getInputRealmLocation() == null || descriptor.getOutputRealmLocation() == null) {
+                continue;
+            }
+            CredentialStore credentialStore;
+            if ((descriptor.getCredentialStore() == null) == (descriptor.getSecretKeyAlias() == null)) {
+                boolean isNew = false;
+                if (descriptor.getCredentialStore() == null) {
```

Comment:
Maybe I'm missing something but in this block, if the above if checks for &&, then we alreayd know `descriptor.getCredentialStore()` is `null` here.